finagle-http/src/main/scala/com/twitter/finagle/http/Request.scala

```diff
@@ -287,10 +287,36 @@ object Request {
    */
   def apply(version: Version, method: Method, uri: String): Request = {
     val reqIn = new DefaultHttpRequest(from(version), from(method), uri)
-    new Request {
-      val httpRequest: HttpRequest = reqIn
-      lazy val remoteSocketAddress: InetSocketAddress = new InetSocketAddress(0)
+    buildRequest(
+      request = reqIn,
+      socketAddress = new InetSocketAddress(0)
+    )
+  }
+
+  private def buildRequest(
+    request: HttpRequest,
+    socketAddress: InetSocketAddress,
+    reader: Option[Reader] = None): Request = {
```

Comment:
Could we remove the default and be explicit? Its private code so I don't see any reason to be ergonomic about it.